airflow/dag_processing/manager.py

```diff
@@ -466,6 +475,12 @@ def __init__(
             else {}
         )
 
+        dags_directory_event_handler = AirflowFileSystemEventHandler(dag_file_processor_manager=self)
+        self.observer = Observer()  # Class watching for file system changes
+        self.observer.schedule(
+            event_handler=dags_directory_event_handler, path=str(self._dag_directory), recursive=True
+        )
```

Comment:
NFS/CIFS shares do not send [inotify](https://man7.org/linux/man-pages/man7/inotify.7.html) events, so I wondering is it possible completely disable watchdog in this case and use current implementation instead?

Reply:
While it would be nice to have this behind a feature flag, `watchdog` will fall back to polling if events are not available. Polling is what we do ourselves so functionally there would not be a difference.

Reply:
I think you'll find IO rate increase on AWS EFS if you did that still. Which is tied to customer billing and throughput available I think.

Reply:
If watchdog doesn't find an API for observing filesystem changes, it will fall back to polling: https://python-watchdog.readthedocs.io/en/stable/installation.html?highlight=polling#supported-platforms-and-caveats. I don't know how much this impacts IO compared to the current implementation, that's something to be tested.